tests/harness/tests/MemoryLoopbackTest/test.cpp

```diff
@@ -0,0 +1,60 @@
+#include <components/MemoryLoopback/MemoryLoopbackImpl.cpp>
+#include <components/MemoryMap/MemoryMapImpl.cpp>
+#include <core/flexus.hpp>
+#include <gtest/gtest.h>
+
+namespace Flexus {
+
+namespace Core {
+void CreateFlexusObject();
+void PrepareFlexusObject();
+void deinitFlexus();
+} // namespace Core
+} // namespace Flexus
+
+// Create fixture for testing the DUT
+class MemoryLoopbackTestFixture : public testing::Test {
+  static MemoryMapConfiguration_struct *aMMCfg;      // A dummy configuration
```

Comment:
Something I don't understand here is why we are creating a dummy memory map in the fixture, instead of relying on the `getMemoryMap()` function and `theMemoryMapFactory` object, which will be called from MemoryLoopbackImpl.initialize()?

Reply:
If I don't create a dummy `MemoryMap`, `theMemoryMapFactory` is not created. Since the factory is by default null, in `MemoryLoopbackImpl.initialize()` I get a segfault. Admittedly I didn't look for an alternative way to initialize the factory. Is it worth spending time on this? Or should I leave it as it is?

Reply:
I am not sure, but I don't think we can have multiple `MemoryMap`s. When I eas playing around with the code, I realised that creating two or more `MemoryMap`s result in some failed assertions. It might be that case that we cannot directly create the factory. But again, I am unsure about this hypothesis.

Reply:
Yes, you are right.
Upon calling `getMemoryMap()`, the control flow goes through the (pre-initialized) factory, which resolves to the function `virtual boost::intrusive_ptr<MemoryMap> createMemoryMap()` implemented by the MemoryMap component.

The right way of doing this for unit testing would be to create a mock component called `MockMemoryMap` which just implements wrappers for two functions that are actually used by MemoryLoopbackImpl:
- `createMemoryMap`
- `recordAccess`

Do you think this is doable for you or shall we just document this and open an issue for it so that we can solve it later?

Reply:
I don't it is doable for me now. I suspect it will take some time. I think we should just open an issue for it.

Reply:
OK. Can you please open that and summarize this convo, so the task state is not lost?

Thanks

Reply:
Sure. I'll do this

Reply:
I've opened this issue. Please have a look